grails-app/controllers/com/unifina/controller/api/PermissionApiController.groovy

```diff
@@ -154,4 +153,30 @@ class PermissionApiController {
 			render status: 204
 		}
 	}
+
+	private usePermissionResource(Class resourceClass, resourceId, Long permissionId, Closure action) {
+		if (!resourceClass) { throw new IllegalArgumentException("Missing resource class") }
+		if (!grailsApplication.isDomainClass(resourceClass)) { throw new IllegalArgumentException("${resourceClass.simpleName} is not a domain class!") }
+		def res = resourceClass.get(resourceId)
+		if (!res) {
+			throw new NotFoundException(resourceClass.simpleName, resourceId.toString())
+		}
+
+		useResource(resourceClass, resourceId, false) { resource ->
+			def p = permissionService.getPermissionsTo(resource).find { it.id == permissionId }
+			if (!p) {
+				throw new NotFoundException("permissions", permissionId.toString())
+			} else {
+				action(p, resource)
+			}
+		}
+	}
+
+	@StreamrApi(authenticationLevel = AuthLevel.NONE)
+	def remove(String id) {
```

Comment:
As described above, let's change `delete` to allow this special case instead of writing a new action.